crates/datasources/src/object_store/azure.rs

```diff
@@ -70,18 +71,38 @@ impl ObjStoreAccess for AzureStoreAccess {
     fn create_store(&self) -> Result<Arc<dyn ObjectStore>> {
         let mut builder = MicrosoftAzureBuilder::new();
 
+        let mut creds = false;
+
         for (key, val) in self.opts.iter() {
+            if matches!(
+                key,
+                AzureConfigKey::AccountName
+                    | AzureConfigKey::AccessKey
+                    | AzureConfigKey::ClientId
+                    | AzureConfigKey::ClientSecret
+                    | AzureConfigKey::SasKey
+                    | AzureConfigKey::Token
+            ) {
+                creds = true;
+            }
+
             builder = builder.with_config(*key, val);
         }
 
         if let Some(account_name) = &self.account_name {
+            creds = true;
             builder = builder.with_account(account_name);
         }
 
         if let Some(access_key) = &self.access_key {
+            creds = true;
             builder = builder.with_access_key(access_key);
         }
 
+        if !creds {
+            builder = builder.with_credentials(Arc::new(NullCredentialProvider));
+        }
```

Comment:
could we create a build with null credentials, and then override it if/as we find credentials, and avoid the bool?